src/analysis/normed_space/conformal.lean

```diff
@@ -0,0 +1,215 @@
+/-
+Copyright (c) 2021 Yourong Zang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yourong Zang
+-/
+import geometry.euclidean.basic
+import geometry.manifold.charted_space
+import analysis.normed_space.inner_product
+
+/-!
+# Conformal Maps on Inner Product Spaces
+
+A map between real inner product spaces `X`, `Y` is `conformal_at` some point `x` in `X` if it is
+real differentiable and its differential at that point is a nonzero constant multiple of
+a linear isometry equivalence.
+
+## Main results
+
+* `conformal_at`: the main definition
+* The conformality of the composition of two conformal maps, the identity map
+  and multiplications by a constant
+* `conformal_at_iff`: an equivalent definition of the conformality at some point
+* `conformal_at_preserves_angle`: if a map is `conformal_at` some point, then its differential
+                                  preserves all angles at that point
+* `conformal_groupoid`: the groupoid of conformal local homeomorphisms
+
+## Tags
+
+conformal
+
+## Warning
+
+The definition of conformality in this file does NOT require the maps to be orientation-preserving.
+Maps such as the conjugate function on the complex plane is considered as conformal maps.
+-/
+
+noncomputable theory
+
+section conformality
+
+/-- A continuous linear map `f'` is said to be conformal if it's
+a nonzero multiple of a bijective linear isometry. -/
+def is_conformal_map {X Y : Type*} [inner_product_space ℝ X] [inner_product_space ℝ Y]
+  (f' : X →L[ℝ] Y) :=
+∃ (c : ℝ) (hc : c ≠ 0) (lie : X ≃ₗᵢ[ℝ] Y), (f' : X → Y) = (λ y, c • y) ∘ lie
+
+/-- A map `f` is said to be conformal if it has a conformal differential `f'`. -/
+def conformal_at {X Y : Type*} [inner_product_space ℝ X] [inner_product_space ℝ Y]
+  (f : X → Y) (x : X) :=
+∃ (f' : X →L[ℝ] Y), has_fderiv_at f f' x ∧ is_conformal_map f'
```

Comment:
If https://github.com/leanprover-community/mathlib/pull/8367/files#r672413402 is sensible, then this would be too:

```suggestion
/-- A map `f` is said to be conformal if it has a conformal differential `f'`. -/
def conformal_at {R X Y : Type*}
  [nondiscrete_normed_field R] [normed_group X] [normed_group Y] [normed_space R X] [normed_space R Y]
  (f : X → Y) (x : X) :=
∃ (f' : X →L[R] Y), has_fderiv_at f f' x ∧ is_conformal_map f'
```

Reply:
There are two generalizations @eric-wieser is proposing.  I'm in favour of generalizing from inner product spaces to normed spaces.

Regarding the generalization from `ℝ` to a more general scalar field:
- For the calculus version (`conformal_at`):  I don't like the idea of requiring an explicit scalar field -- the `ℝ`-case is by far the most important (if not the only) use case, and it would be a real pain to have to say `conformal_at ℝ ...` all the time.
- For the linear algebra version (`is_conformal_map`):  I don't have a strong opinion about whether to generalize the scalar field or not.